frontend/src/pages/cabins/admin/index.tsx

```diff
@@ -51,15 +42,28 @@ const AdminPage: NextPage = () => {
   const { data, refetch } = useQuery<{
     adminAllBookings: BookingFromQuery[];
   }>(QUERY_ADMIN_ALL_BOOKINGS, { variables: { after: dayjs().subtract(1, "day").format("YYYY-MM-DD") } });
-  const [confirmBooking] = useMutation(CONFIRM_BOOKING, { refetchQueries: [{ query: QUERY_ADMIN_ALL_BOOKINGS }] });
   const [openSnackbar, setOpenSnackbar] = useState(false);
   const [snackbarMessage, setSnackbarMessage] = useState("");
   const [bookingToBeDeleted, setBookingToBeDeleted] = useState<BookingFromQuery | undefined>();
-  const [sendEmail] = useMutation(SEND_EMAIL);
+  const [accepted, setAccepted] = useState<BookingFromQuery[]>();
+  const [declined, setDeclined] = useState<BookingFromQuery[]>();
+  const [tentative, setTentative] = useState<BookingFromQuery[]>();
+  const [tabValue, setTabValue] = useState<number>(0);
+  const router = useRouter();
+
+  const handleTabChange = (newTabValue: number) => setTabValue(newTabValue);
 
   const isMobile = useMediaQuery(theme.breakpoints.down("sm"));
   const classes = useStyles();
 
+  useEffect(() => {
+    if (data?.adminAllBookings) {
+      setAccepted(data.adminAllBookings.filter((booking) => !booking.isTentative && !booking.isDeclined));
+      setDeclined(data.adminAllBookings.filter((booking) => !booking.isTentative && booking.isDeclined));
+      setTentative(data.adminAllBookings.filter((booking) => booking.isTentative));
+    }
+  }, [data]);
```

Comment:
These could all be
```typescript
const accepted = data?.adminAllBookings.filter((booking) => !booking.isTentative && !booking.isDeclined)
```
As the Apollo queries are stateful, and any changes to them will trigger a rerender.